File: cloud-storage-spring-java/src/main/java/com/github/migangqui/cloud/storage/spring/service/StorageService.java
package com.github.migangqui.cloud.storage.spring.service;

import com.github.migangqui.cloud.storage.spring.bean.*;

import java.io.InputStream;
import java.util.concurrent.Future;

public interface StorageService {

	UploadFileResponse uploadFile(UploadFileRequest request);

	Future<UploadFileResponse> uploadFileAsync(UploadFileRequest request);

	GetFileResponse getFile(GetFileRequest request);

	DeleteFileResponse deleteFile(DeleteFileRequest request);

}

File: cloud-storage-spring-kotlin/src/main/kotlin/com/github/migangqui/cloud/storage/spring/service/StorageService.kt
package com.github.migangqui.cloud.storage.spring.service

import com.github.migangqui.cloud.storage.spring.bean.*
import java.util.concurrent.Future

interface StorageService {
    fun uploadFile(request: UploadFileRequest): UploadFileResponse

    fun uploadFileAsync(request: UploadFileRequest): Future<UploadFileResponse>

    fun getFile(request: GetFileRequest): GetFileResponse

    fun deleteFile(request: DeleteFileRequest): DeleteFileResponse
}